path-traversal/COP4331-Large-Project_FoodGram/api.js

```
@@ -0,0 +1,611 @@
+
+exports.setApp = function (app ) {
+
+app.post('/api/register/', async (req, res, next) =>
+{
+  // incoming: firstName, lastName, login, password
+  // outgoing: error
+  const { FirstName, LastName , Login , Password , Email} = req.body;
+  console.log(FirstName,LastName);
+  console.log(User.findOne({Login: Login}))
+  var user = await User.findOne({Login: Login})
+  if (user)
```

Biome (2.1.2)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)